WWDC/PlaybackViewModel.swift

```diff
@@ -126,6 +126,17 @@ final class PlaybackViewModel {
 
                 if !d.isZero { self.nowPlayingInfo.value?.progress = p / d }
             }
+
+            NotificationCenter.default.addObserver(self, selector: #selector(self.playerDidReachEndOfVideo(note:)), name: NSNotification.Name.AVPlayerItemDidPlayToEndTime, object: nil)
```

Comment:
This notification only fires if they play to the very, _very_ end, which won't necessarily align with the other notions we have of "watched". We also probably shouldn't delete the video that is currently being played. Are there any other integration points for a feature like this?

Reply:
Yeah, it should be deleted when the app considers it as being watched. And there's also the consideration of when to actually delete the video, since deleting the video file that's currently being played sounds dangerous.

Reply:
Another question is: should we delete the video if the user manually marks it as watched?

Reply:
Now I'm starting to wonder if this feature is even worth it. Is it really that hard to press a button after you finish watching the video? Maybe we could display the delete option more prominently over the video when it's near the end? 🤔

Reply:
Post roll overlay? Like on Net... uh, major video subscription services ;-)

Reply:
If we went a fully automatic route I'd be in favor of a cleanup job on app launch. Also, just confirmed users can delete the currently playing video manually already 😂😭

Reply:
What happens if you do? 😛

Reply:
Everything works fine. I'm guessing due to having a robust OS and frameworks that don't let the file get completely removed when it's open? Or maybe AVFoundation defensive programming?